dascore/utils/patch.py

```
@@ -1799,20 +1799,24 @@ def _joinable(coords, dim: str) -> list[np.ndarray]:
+    zeros or empty labels.
+
+    Blankness is a question about values, not about type: a coordinate
+    which states no values is blank whether it remembers being made of
+    times or has forgotten.
     """
     arrays = [x.values for x in coords]
-    blank = [x.dtype.kind == "f" and bool(np.all(pd.isnull(x))) for x in arrays]
+    blank = [bool(np.all(pd.isnull(x))) for x in arrays]
     if all(blank) or not any(blank):
         return arrays
     target = np.result_type(*[x.dtype for x, b in zip(arrays, blank) if not b])
```

Ruff (0.16.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)